cmake/Dependencies.cmake

```diff
@@ -1216,6 +1216,26 @@ if(USE_GLOO)
   endif()
 endif()
 
+if(USE_DISTRIBUTED AND USE_TENSORPIPE)
+  if(MSVC)
+    message(WARNING "Tensorpipe cannot be used on Windows.")
+  else()
+    set(__PYTORCH_BUILD ${PYTORCH_BUILD})
+    set(PYTORCH_BUILD ON)
```

Comment:
Question: (it is just me not familiar with cmake), looks like we are setting and restoring `PYTORCH_BUILD` and `BUILD_TESTING`, will this be a problem if there are concurrent threads building other modules?

Reply:
@mrshenli , it's current value is saved on the previous line and restored on line 1232

Reply:
This only happens during the parsing of CMake files, which AFAIK is done in one thread.